mysql-test/suite/innodb/t/mem_pressure.test

```diff
@@ -41,4 +41,35 @@ let SEARCH_PATTERN= InnoDB: Memory pressure event freed.*;
 
 set debug_dbug=@save_dbug;
 
+--echo #
+--echo # MDEV-34753 memory pressure - erroneous termination condition
+--echo #
+
+# Just do the same thing again. Previously before MDEV-34753 the handling
+# would have quit with no more Memory pressure events.
+CREATE TABLE t1 (a INT PRIMARY KEY) ENGINE=InnoDB;
+SET GLOBAL innodb_limit_optimistic_insert_debug=2;
+SET STATEMENT unique_checks=0, foreign_key_checks=0 FOR
+INSERT INTO t1 SELECT * FROM seq_1_to_1000;
+
+SET GLOBAL innodb_limit_optimistic_insert_debug=@save_limit;
+DROP TABLE t1;
+
+SELECT CAST(VARIABLE_VALUE AS INTEGER) INTO @dirty_prev
+FROM INFORMATION_SCHEMA.GLOBAL_STATUS
+WHERE VARIABLE_NAME='Innodb_buffer_pool_pages_dirty';
+
+set debug_dbug="d,trigger_garbage_collection";
+SET GLOBAL innodb_buffer_pool_size=@@innodb_buffer_pool_size;
+
+SELECT CAST(VARIABLE_VALUE AS INTEGER) < @dirty_prev AS LESS_DIRTY_IS_GOOD
+FROM INFORMATION_SCHEMA.GLOBAL_STATUS
+WHERE VARIABLE_NAME='Innodb_buffer_pool_pages_dirty';
+
+let SEARCH_FILE= $MYSQLTEST_VARDIR/log/mysqld.1.err;
+let SEARCH_PATTERN= InnoDB: Memory pressure event freed.*;
+--source include/search_pattern_in_file.inc
+
+set debug_dbug=@save_dbug;
+
```

Comment:
Well, if you really want to run this twice, do it in a while ($i < 2) { ... inc $i; } loop, don't copy-paste duplicate the code.

Or maybe it's fine to run it just once as originally, if you fix the debug code that triggers the condition from debug injection. That seems to have been the reason that this test could succeed, even though the code in the thread could never run.

But you still need to fix the race condition in this test, see other comment.